atlasdb-ete-tests/src/test/java/com/palantir/atlasdb/ete/LockWatchEteTest.java

```diff
@@ -0,0 +1,265 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.ete;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.common.collect.ImmutableSet;
+import com.palantir.atlasdb.encoding.PtBytes;
+import com.palantir.atlasdb.keyvalue.api.AtlasLockDescriptorUtils;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.lock.EteLockWatchResource;
+import com.palantir.atlasdb.lock.GetLockWatchUpdateRequest;
+import com.palantir.atlasdb.lock.SimpleEteLockWatchResource;
+import com.palantir.atlasdb.lock.TransactionId;
+import com.palantir.atlasdb.lock.WriteRequest;
+import com.palantir.lock.AtlasRowLockDescriptor;
+import com.palantir.lock.LockDescriptor;
+import com.palantir.lock.watch.CommitUpdate;
+import com.palantir.lock.watch.CommitUpdate.Visitor;
+import com.palantir.lock.watch.LockEvent;
+import com.palantir.lock.watch.LockWatchCreatedEvent;
+import com.palantir.lock.watch.LockWatchEvent;
+import com.palantir.lock.watch.LockWatchVersion;
+import com.palantir.lock.watch.TransactionsLockWatchUpdate;
+import com.palantir.lock.watch.UnlockEvent;
+import java.util.Collection;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.Before;
+import org.junit.Test;
+
+public final class LockWatchEteTest {
+    private static final String SEED = "seed";
+    private static final String ROW_1 = row(1);
+    private static final String ROW_2 = row(2);
+    private static final String ROW_3 = row(3);
+
+    private final EteLockWatchResource lockWatcher = EteSetup.createClientToSingleNode(EteLockWatchResource.class);
+
+    private TableReference tableReference;
+
+    @Before
+    public void before() {
+        String tableName = UUID.randomUUID().toString().substring(0, 16).replace("-", "X");
```

Comment:
nit: standard in other tests is to replace with `_`, not `X`.
```suggestion
        String tableName = UUID.randomUUID().toString().substring(0, 16).replace("-", "_");
```